packages/plugins/plugin-device-token/src/DeviceTokenPlugin.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  PlatformPlugin,
+  SegmentClient,
+  PluginType,
+} from '@segment/analytics-react-native';
+import { Platform } from 'react-native';
+import messaging from '@react-native-firebase/messaging';
+
+export class DeviceTokenPlugin extends PlatformPlugin {
+  type = PluginType.enrichment;
+
+  async configure(analytics: SegmentClient) {
+    const isAuthorized = await this.checkUserPermission();
```

Comment:
You can still do this with the promise variable you had before, that was still a good idea as the value is retrieved on construction time, then just awaited on `configure`.
By my previous comment I just meant that promises are a value. So you they won't change, but that still makes it useful here.

Reply:
oh okay! That makes complete sense thank you